plugins/rest/aws.go

```diff
@@ -470,16 +497,16 @@ func signV4(req *http.Request, credService awsCredentialService, theTime time.Ti
 	// required format of Authorization header; n.b. the access key corresponding to
 	// the secret key is included here
 	authHdr := "AWS4-HMAC-SHA256 Credential=" + creds.AccessKey + "/" + dateNow
-	authHdr += "/" + creds.RegionName + "/s3/aws4_request,"
+	authHdr += "/" + creds.RegionName + "/" + service + "/aws4_request,"
 	authHdr += "SignedHeaders=" + headerList + ","
 	authHdr += "Signature=" + fmt.Sprintf("%x", signature)
 
 	// add the computed Authorization
-	req.Header.Add("Authorization", authHdr)
+	req.Header.Set("Authorization", authHdr)
 
 	// populate the other signed headers into the request
 	for _, k := range orderedKeys {
-		req.Header.Add(k, awsHeaders[k])
+		req.Header.Set(k, awsHeaders[k])
```

Comment:
I don't think we ever use them in this context, but this would overwrite multi-value headers in the request if present:
```
Accept: text/plain
Accept: text/html
```
Any particular reason to change this? For `Authorization` above it's fine though as there should be no more than one such header.

Reply:
I swapped to `Set` since we add the existing headers to `awsHeaders` for them all to be signed, and `Add` would duplicate those existing values. I'll keep track of the new and existing headers separately so we don't overwrite any multi-value headers :)

Reply:
Awesome!